common/cuda_hip/base/device_matrix_data_kernels.hpp.inc

```diff
@@ -39,12 +39,12 @@ void remove_zeros(std::shared_ptr<const DefaultExecutor> exec,
     // due to overly generic assignment operator and constructor leading to
     // ambiguities. So we need to use our own fake_complex type
     using device_value_type = device_member_type<ValueType>;
-    auto value_ptr = thrust::device_pointer_cast(
-        reinterpret_cast<const device_value_type*>(values.get_const_data()));
+    auto value_ptr =
+        reinterpret_cast<const device_value_type*>(values.get_const_data());
```

Comment:
Do we not need the thrust pointer cast any more because of the new execution policy, or was it always unnecessary? Either way, it is nice that we can do without it.

Reply:
it was always unnecessary, I only had some segfaults in the past that made me think it was necessary, but the issue must have been elsewhere